sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -0,0 +1,211 @@
+                DEFAULT_REDIRECT_LOCATION_HEADER_NAME));
+            this.locationHeader = DEFAULT_REDIRECT_LOCATION_HEADER_NAME;
+        } else {
+            this.locationHeader = locationHeader;
+        }
+        if (CoreUtils.isNullOrEmpty(allowedMethods)) {
+            logger.error(String.format("'allowedMethods' provided as null will be defaulted to %s",
+                DEFAULT_REDIRECT_ALLOWED_METHODS));
+            this.redirectMethods = DEFAULT_REDIRECT_ALLOWED_METHODS;
+        } else {
+            this.redirectMethods = allowedMethods;
+        }
+    }
+
+    @Override
+    public boolean shouldAttemptRedirect(HttpPipelineCallContext context,
+                                         HttpResponse httpResponse, int tryCount,
+                                         Set<String> attemptedRedirectUrls) {
+        String redirectUrl =
+            tryGetRedirectHeader(httpResponse.getHeaders(), this.getLocationHeader());
+
+        if (isValidRedirectCount(tryCount)
+            && redirectUrl != null
+            && !alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)
+            && isValidRedirectStatusCode(httpResponse.getStatusCode())
+            && isAllowedRedirectMethod(httpResponse.getRequest().getHttpMethod())) {
+            logger.verbose("[Redirecting] Try count: {}, Attempted Redirect URLs: {}", tryCount,
+                attemptedRedirectUrls.toString());
+            attemptedRedirectUrls.add(redirectUrl);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public HttpRequest createRedirectRequest(HttpResponse httpResponse) {
+        String responseLocation =
+            tryGetRedirectHeader(httpResponse.getHeaders(), this.getLocationHeader());
+        return httpResponse.getRequest().setUrl(responseLocation);
+    }
+
+    @Override
+    public int getMaxAttempts() {
+        return maxAttempts;
+    }
+
+    @Override
+    public String getLocationHeader() {
+        return locationHeader;
+    }
+
+    @Override
+    public Set<HttpMethod> getAllowedMethods() {
+        return redirectMethods;
+    }
+
+    /**
+     * Check if the redirect url provided in the response headers is already attempted.
+     *
+     * @param redirectUrl the redirect url provided in the response header.
+     * @param attemptedRedirectUrls the set containing a list of attempted redirect locations.
+     * @return {@code true} if the redirectUrl provided in the response header is already being attempted for redirect
+     * , {@code false} otherwise.
+     */
+    private boolean alreadyAttemptedRedirectUrl(String redirectUrl,
+                                                Set<String> attemptedRedirectUrls) {
+        if (attemptedRedirectUrls.contains(redirectUrl)) {
+            logger.error(String.format("Request was redirected more than once to: %s", redirectUrl));
```

Comment:
This isn't a great way to do logging - you should do it without the `String.format` call, as you're pre-emptively creating the string even if the logging is not going to happen.